### PR TITLE
Enable StackGres API

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

Enable the StackGres restAPI and add a networkpolicy to allow access to it.

This also changes the `golden-diff` target a bit, as it would always find changes in the generated password.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/main.jsonnet [hunk 1]
```diff
@@ -5,7 +5,37 @@ local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.stackgres_operator;
 
+local networkPolicy = kube.NetworkPolicy('allow-stackgres-api') + {
```
Comment (zugao) [inline component/main.jsonnet:8]: Are we sure network policies are to be managed here in this component? Have you had a discussion with @vshn/aldebaran ?

**Review by zugao — APPROVED** [review]
LGTM

File: component/main.jsonnet (hunk 1 above)
Comment (simu) [reply component/main.jsonnet:8]: Yes, components are responsible for setting up appropriate network policies if there's a need for more broad access than what you get by default (Project Syn default network policies basically isolate namespaces, cf. https://github.com/projectsyn/component-networkpolicy/blob/master/component/main.jsonnet).

Comment (Kidswiss) [reply component/main.jsonnet:8]: Thanks!

**Pull request merged**